src/c_api/c_api.cu

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) 2019 by Contributors
+
+#include <thrust/device_ptr.h>
+#include <thrust/device_vector.h>
+#include <thrust/execution_policy.h>
+#include <thrust/scan.h>
+#include <xgboost/c_api.h>
+#include <xgboost/data.h>
+#include <memory>
+
+#include "../common/device_helpers.cuh"
+#include "../common/host_device_vector.h"
+#include "../data/simple_csr_source.h"
+#include "./c_api_error.h"
+
+namespace xgboost {
+typedef unsigned char cudf_interchange_valid_type;
+typedef int32_t
+    cudf_interchange_size_type;  // Limits the maximum size of a
+                                 // cudf_interchange_column to 2^31-1
+typedef enum {
+  invalid = 0,
+  INT8,
+  INT16,
+  INT32,
+  INT64,
+  FLOAT32,
+  FLOAT64,
+} cudf_interchange_dtype;

```

Comment:
Restricting to numeric types is a good idea. Those are not likely to change any time soon.